### DISCUSSION TITLE
Invalid Literal For int() With Base.. Error

### DISCUSSION THREAD

**Discussion by omerbykbs** [community/community#21489] [Other Feature Feedback, Questions, & Ideas]
<p>Hello! I got stuck in this error! I can’t figure it out what exactly wrong with this code. Could you explain me, please?</p>
<p></p><div class="lightbox-wrapper"><a class="lightbox" href="https://aws1.discourse-cdn.com/github/original/2X/5/54000463d3a843d5e859b10eeef345fbaf907918.jpeg" data-download-href="/uploads/short-url/bZ68WwDs5XCSR8qTAVRcz6CG0kM.jpeg?dl=1" title="Python 1" rel="nofollow noopener"><img src="https://user-images.githubusercontent.com/68295684/181064996-61381249-3408-4999-a25e-a0e5e8db73ba.jpeg" alt="Python 1" data-base62-sha1="bZ68WwDs5XCSR8qTAVRcz6CG0kM" width="690" height="387" srcset="https://aws1.discourse-cdn.com/github/optimized/2X/5/54000463d3a843d5e859b10eeef345fbaf907918_2_690x387.jpeg, https://aws1.discourse-cdn.com/github/optimized/2X/5/54000463d3a843d5e859b10eeef345fbaf907918_2_1035x580.jpeg 1.5x, https://aws1.discourse-cdn.com/github/original/2X/5/54000463d3a843d5e859b10eeef345fbaf907918.jpeg 2x" data-small-upload="https://aws1.discourse-cdn.com/github/optimized/2X/5/54000463d3a843d5e859b10eeef345fbaf907918_2_10x10.png"><div class="meta">
<svg class="fa d-icon d-icon-far-image svg-icon" aria-hidden="true"><use xlink:href="#far-image"></use></svg><span class="filename">Python 1</span><span class="informations">1366×768 160 KB</span><svg class="fa d-icon d-icon-discourse-expand svg-icon" aria-hidden="true"><use xlink:href="#discourse-expand"></use></svg>
</div></a></div>

**Comment by airtower-luna** [comment]
<p>Could you explain what you are trying to do? The immediate problem is that the <code>grade</code> string is empty.</p>
<p>At a larger scope, it looks like you expect the <code>input('text here')</code> calls to return that text. That’s not going to work, input uses the text as a prompt and returns whatever you type in response. I guess you actually want to use those texts as your sample text, so instead you could use e.g. for grades:</p>
<pre><code class="lang-auto">grades = '81, 77, 92, 88'.split(', ')
</code></pre>
<p>Though it’d make more sense to skip the split and directly use a list:</p>
<pre><code class="lang-auto">grades = [81, 77, 92, 88]
</code></pre>

**Comment by omerbykbs** [comment]
<p>Firstly I am so thankful for your response <a class="mention" href="/airtower-luna">@airtower-luna</a><br>
What am I trying to do: I am just a beginner in Python and try to figure it out how the codes work. Actually I’m doing an Online Course from a website and that code was written as a solution for a quiz from Scripting with Raw Input Subject. But as I tried to write it and to see how it works, I was just surprised because of that “Error”.</p>
<p>I just tried what you told me that just using directly a list but it didn’t help me. I couldn’t see still the output like in the solution below:</p>
<p></p><div class="lightbox-wrapper"><a class="lightbox" href="https://aws1.discourse-cdn.com/github/original/2X/8/82a438178ce44a83720b713d0d4999181400d706.jpeg" data-download-href="/uploads/short-url/iDHUm6MXMZBwu0vQPNxq1eMGL4y.jpeg?dl=1" title="Problem3" rel="nofollow noopener"><img src="https://user-images.githubusercontent.com/68295684/181064997-b164af6c-7de6-436c-b6b1-54945a8b4f0a.jpeg" alt="Problem3" data-base62-sha1="iDHUm6MXMZBwu0vQPNxq1eMGL4y" width="690" height="368" srcset="https://aws1.discourse-cdn.com/github/optimized/2X/8/82a438178ce44a83720b713d0d4999181400d706_2_690x368.jpeg, https://aws1.discourse-cdn.com/github/original/2X/8/82a438178ce44a83720b713d0d4999181400d706.jpeg 1.5x, https://aws1.discourse-cdn.com/github/original/2X/8/82a438178ce44a83720b713d0d4999181400d706.jpeg 2x" data-small-upload="https://aws1.discourse-cdn.com/github/optimized/2X/8/82a438178ce44a83720b713d0d4999181400d706_2_10x10.png"><div class="meta">
<svg class="fa d-icon d-icon-far-image svg-icon" aria-hidden="true"><use xlink:href="#far-image"></use></svg><span class="filename">Problem3</span><span class="informations">875×467 87.8 KB</span><svg class="fa d-icon d-icon-discourse-expand svg-icon" aria-hidden="true"><use xlink:href="#discourse-expand"></use></svg>
</div></a></div>

**Comment by airtower-luna** [comment]
<p>Well, if you want help you’ll have to describe the actual problem: what are you doing (in this case: code!), what is happening, what do you expect to happen instead? That screenshot covers only the last part.</p>

**Comment by omerbykbs** [comment]
<p>Sorry I have tried to put two more photos but I couldn’t because I am just a new member, I am not allowed to put more photos. But I copied and pasted below what the problem is:</p>
<h1>Quiz: Generate Messages</h1>
<p>Imagine you’re a teacher who needs to send a message to each of your students reminding them of their missing assignments and grade in the class. You have each of their names, number of missing assignments, and grades on a spreadsheet and just have to insert them into placeholders in this message you came up with:</p>
<blockquote>
<p>Hi  <strong>[insert student name]</strong> ,</p>
</blockquote>
<blockquote>
<p>This is a reminder that you have  <strong>[insert number of missing assignments]</strong>  assignments left to submit before you can graduate. Your current grade is  <strong>[insert current grade]</strong>  and can increase to  <strong>[insert potential grade]</strong>  if you submit all assignments before the due date.</p>
</blockquote>
<p>You can just copy and paste this message to each student and manually insert the appropriate values each time, but instead you’re going to write a program that does this for you.</p>
<p>Write a script that does the following:</p>
<ol>
<li>Ask for user input 3 times. Once for a list of names, once for a list of missing assignment counts, and once for a list of grades. Use this input to create lists for  <code>names</code> ,  <code>assignments</code> , and  <code>grades</code> .</li>
<li>Use a loop to print the message for each student with the correct values. The potential grade is simply the current grade added to two times the number of missing assignments.</li>
</ol>
<h3>Template code for your script:</h3>
<pre><code class="lang-auto">names =  # get and process input for a list of names
assignments =  # get and process input for a list of the number of assignments
grades =  # get and process input for a list of grades

# message string to be used for each student
# HINT: use .format() with this string in your for loop
message = "Hi {},\n\nThis is a reminder that you have {} assignments left to \
submit before you can graduate. You're current grade is {} and can increase \
to {} if you submit all assignments before the due date.\n\n"

# write a for loop that iterates through each set of names, assignments, and grades to print each student's message

</code></pre>

**Comment by fire-eggs** [comment, marked as answer]
<p>What <a class="mention" href="/airtower-luna">@airtower-luna</a> and others are looking for is the code that you’ve currently written. Copy it from your terminal / editor / browser, and paste it as a reply here using the “preformatted text” icon. We don’t want screenshots, we want text.</p>
<p>Based on your template code, the program needs to prompt for input, and you, as the user, must manually enter the data. As <a class="mention" href="/airtower-luna">@airtower-luna</a> points out, you need to use <code>input</code> correctly, e.g.:</p>
<pre><code class="lang-auto">grades = input('enter grades')
</code></pre>
<p>You then would type in the necessary string, i.e. <code>81,77,92,88</code>.</p>

**Comment by omerbykbs** [comment]
<p>Right now I got it! It was my foolishness that I thought the code gets the inputs automatically… Thank you so much both <a class="mention" href="/fire-eggs">@fire-eggs</a> and <a class="mention" href="/airtower-luna">@airtower-luna</a></p>

**Comment by fire-eggs** [comment]
<p>Don’t beat yourself up about this! Not foolishness - ignorance. Ignorance can be corrected, you are learning and we all had to learn this stuff some time!</p>
